src/dom.js

```diff
@@ -87,14 +88,36 @@ export function waitForChildPromise(parent, checkFunc) {
 }
 
 /**
- * Waits for document's body to be available.
+ * Waits for document's head to be available.
+ * @param {!Document} doc
+ * @param {function()} callback
+ */
+export function waitForHead(doc, callback) {
+  waitForChild(doc.documentElement, () => !!doc.head, callback);
```

Comment:
We should be waiting (ironically) for `doc.body` here, not `doc.head`. The presence of `doc.body` means that the `<head>` (and all its descendants) have been parsed. Waiting only for `doc.head` means the `<head>` tag's opening tag has been seen, but the descendants may not be fully parsed yet.